src/core/source_buffers/queued_source_buffer.ts

```diff
@@ -584,27 +584,62 @@ function convertQueueItemToTask<T>(
       // to be separated into two steps
       const steps = [];
       const itemValue = item.value;
-      const { data } = itemValue;
+      const { data, inventoryInfos } = itemValue;
+      const { estimatedDuration, estimatedStart, segment } = inventoryInfos;
+
+      // Cutting exactly at the start or end of the appendWindow can lead to
+      // cases of infinite rebuffering due to how browser handle such windows.
+      // To work-around that, we add a small offset before and after those.
+      const safeAppendWindow : [ number | undefined, number | undefined ] = [
+        data.appendWindow[0] !== undefined ?
+          Math.max(0, data.appendWindow[0] - APPEND_WINDOW_SECURITIES.START) :
+          undefined,
+        data.appendWindow[1] !== undefined ?
+          data.appendWindow[1] + APPEND_WINDOW_SECURITIES.END :
+          undefined,
+      ];
+
       if (data.initSegment !== null) {
         steps.push({ isInit: true,
                      segmentData: data.initSegment,
                      codec: data.codec,
                      timestampOffset: data.timestampOffset,
-                     appendWindow: data.appendWindow });
+                     appendWindow: safeAppendWindow });
       }
       if (data.chunk !== null) {
         steps.push({ isInit: false,
                      segmentData: data.chunk,
                      codec: data.codec,
                      timestampOffset: data.timestampOffset,
-                     appendWindow: data.appendWindow });
+                     appendWindow: safeAppendWindow });
       }
       if (steps.length === 0) {
         return null;
       }
+
+      let start = estimatedStart === undefined ? segment.time / segment.timescale :
+                                                 estimatedStart;
+      const duration = estimatedDuration === undefined ?
+        segment.duration / segment.timescale :
+        estimatedDuration;
+      let end = start + duration;
+
+      if (safeAppendWindow[0] !== undefined) {
```

Comment:
Should `duration` adapt to new start and end ?

Reply:
duration is not used anymore and should define the duration of the segment pre-appendWindows. So I don't see the point to adapt `duration`